src/com/alevel/cw7/actions/Shop.java

```diff
@@ -0,0 +1,70 @@
+package com.alevel.cw7.actions;
+
+import com.alevel.cw7.models.Player;
+import com.alevel.cw7.models.Present;
+
+import java.util.Random;
+import java.util.Scanner;
+
+public class Shop {
+
+    private final Random RAND = new Random();
+
+    public void doAction(Player player, int playerChoose) {
+        Present[] variantsPresent = Present.values();
+        int variant = RAND.nextInt(variantsPresent.length);
+        int price;
+
+        switch (playerChoose) {
+            case 0: {
+                price = 100;
+                if (player.getMoney() < price) {
+                    System.out.printf("%s have no %d coins\n", player.getName(), price);
+                    break;
+                }
+                player.updateMoney(-price);
+                player.setPresents(addPresent(player, variantsPresent[variant]));
+                player.updateHappiness(20);
+                System.out.printf("%s received a %s as a gift\n", player.getName(), variantsPresent[variant]);
+                break;
+            }
+            case 1: {
+                price = 150;
+                if (player.getMoney() < price) {
+                    System.out.printf("%s have no %d coins\n", player.getName(), price);
+                    break;
+                }
+                player.updateMoney(-price);
+                player.setPresents(addPresent(player, variantsPresent[variant]));
+                player.updateHappiness(25);
+                System.out.printf("%s received a %s as a gift\n", player.getName(), variantsPresent[variant]);
+                break;
+            }
+            case 2: {
+                price = 200;
+                if (player.getMoney() < price) {
+                    System.out.printf("%s have no %d coins\n", player.getName(), price);
+                    break;
+                }
+                player.updateMoney(-price);
+                player.setPresents(addPresent(player, variantsPresent[variant]));
+                player.updateHappiness(30);
+                System.out.printf("%s received a %s as a gift\n", player.getName(), variantsPresent[variant]);
+                break;
+            }
+            default: {
+                System.out.println("Enter number between 0 and 2");
+            }
+        }
+    }
+
+    private Present[] addPresent(Player player, Present item) {
+        Present[] tmp = new Present[player.getPresents().length + 1];
```

Comment:
I would advise you to make an increase according to the formula + player.getPresents().length / 2 (starting value must be 2 or more)